inspirehep/modules/literaturesuggest/dojson/fields/literature.py

```diff
@@ -215,10 +217,11 @@ def year(self, key, value):
 
 @literature.over('_page_range_article_id', '^page_range_article_id$')
 def page_range_article_id(self, key, value):
-    if 'publication_info' in self:
-        self['publication_info'][0].update(dict(page_artid=value))
-    else:
-        self['publication_info'] = [dict(page_artid=value)]
+    page_start, page_end, artid = split_page_artid(value)
+    self.setdefault('publication_info', [{}]).update(dict(
```

Comment:
Uh? `setdefault` should return the second argument, which then doesn't have an `update` method defined on it because it is a list (and not a dict).